src/core/packet.lua

```diff
@@ -109,3 +114,21 @@ function preallocate_step()
    packet_allocation_step = 2 * packet_allocation_step
 end
 
+function dump(p, w)
```

Comment:
Likewise the dump function is probably not needed at all in core.packet, and we probably shouldn't be adding the metatype here -- or if we do, we should document it.  I believe these changes are from @javierguerragiraldez originally, and though they are likely very nice changes it will simplify our lives to not include them in this PR.  Seems to me, anyway; wdyt?